### PR TITLE
Code Review Request

### PR DESCRIPTION
**Pull request by CiaranGruber**
@Keegan-Cruickshank. Please check for possible refactoring and other improvements

### CODE REVIEW

File: prac_06/car.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+
+class Car:
+    """Represent a Car object."""
+
+    def __init__(self, fuel=0):
+        """Initialise a Car instance.
+
+        fuel: float, one unit of fuel drives one kilometre
+        """
```
Comment (Keegan-Cruickshank) [inline prac_06/car.py]: I think the multi line comment needs starting quotes and finishing quotes on a new line, What's your thoughts?

File: prac_06/used_cars.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    my_car.drive(30)
+    print("fuel =", my_car.fuel)
+    print("odo =", my_car.odometer)
+    print(my_car)
+
+    print("Car {}, {}".format(my_car.fuel, my_car.odometer))
+    print("Car {self.fuel}, {self.odometer}".format(self=my_car))
+
+
+main()
```
Comment (Keegan-Cruickshank) [inline prac_06/used_cars.py]: PEP 8 Guidelines like a new line at the end of a python file :)

File: prac_06/programming_language.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+        self.name = name
+        self.typing = typing
+        self.reflection = reflection
+        self.year = year
+
+    def __str__(self):
+        return '{}, {} Typing, Reflection={}, First appeared in {}'.format(self.name, self.typing, self.reflection,
+                                                                           self.year)
+
+    def is_dynamic(self):
```
Comment (Keegan-Cruickshank) [inline prac_06/programming_language.py]:
```python
def is_dynamic(self) -> bool:
    if self.typing == "Dynamic":
        return True
    else:
        return False
```

Try this for your is_dynamic function

File: prac_06/languages.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+def main():
+    ruby = ProgrammingLanguage("Ruby", "Dynamic", True, 1995)
+    python = ProgrammingLanguage("Python", "Dynamic", True, 1991)
+    visual_basic = ProgrammingLanguage("Visual Basic", "Static", False, 1991)
+    print(ruby)
+    print()
+    print('The dynamic typed languages are:')
+    for programming_language in [ruby, python, visual_basic]:
+        if programming_language.typing == 'Dynamic':
```
Comment (Keegan-Cruickshank) [inline prac_06/languages.py]: Now rather then inline testing you can use my comment below to say:
```python
if programming_language.is_dynamic:
```

File: prac_06/guitars.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        cost = get_number('Cost: ', 'Year must be a number', 'Year must be above 0', 0)
+        guitars.append(Guitar(name, year, cost))
+        print()
+        name = input('Name: ')
+
+    print()
+    longest_word = get_largest_name(guitars)
+    longest_cost = get_largest_cost(guitars)
+    for i, guitar in enumerate(guitars):
+        print("Guitar {0}: {self.name:>{1}} ({self.year}), worth ${self.cost:{2},.2f}{3}"
```
Comment (Keegan-Cruickshank) [inline prac_06/guitars.py]: Personally I find this hard to follow using default naming seems cleaner in this case.

Comment (Keegan-Cruickshank) [reply prac_06/guitars.py]:
```python
    for i, guitar in enumerate(guitars):
        print(
            "Guitar {}: {} ({}), worth ${:0,.2f}"
            .format(i + 1, guitar.name, guitar.year, guitar.cost),
            end=""
        )
        if guitar.is_vintage():
            print(" (vintage)")
        else:
            print("")
```

**Review by Keegan-Cruickshank — COMMENTED** [review]
Changes to guitar.py

File: prac_06/car.py (hunk 1 above)
Comment (CiaranGruber) [reply prac_06/car.py]: I'm not too certain though I think some description docstrings need to have no space at the start. As this entire file is just the default (except for the def __str__), I'll probably keep it just in case

File: prac_06/used_cars.py (hunk 2 above)
Comment (CiaranGruber) [reply prac_06/used_cars.py]: Added to the file 👍

File: prac_06/programming_language.py (hunk 3 above)
Comment (CiaranGruber) [reply prac_06/programming_language.py]: Added the return type however I have kept the main part of the code as it was recommended (or at the very least shown as an example) in previous lectures

File: prac_06/languages.py (hunk 4 above)
Comment (CiaranGruber) [reply prac_06/languages.py]: Seems like I had already fixed it later. Must have created it after the pull request but all fixed anyway 👍
